ratelimits/limiter.go

```diff
@@ -168,11 +183,24 @@ func (l *Limiter) Spend(ctx context.Context, name Name, id string, cost int64) (
 		return nil, ErrInvalidCostOverLimit
```

Comment:
1. I don't like transforming an invalid request into a "Denied" decision. We actually plan use those denied decisions to inform various meta-data in the headers of the returned 429 response.

2. This should essentially never happen. We are the ones who decide both the cost and the limit.

Reply:
This makes sense, but is it truly an invalid request if the client has a "pending authorizations" Burst limit override of 50, but they send a newOrder request with 100 names? That seems to me like it should simply be a denial, not some other form of error.

Reply:
From our discussion offline: if we configured a rate limit such that this is occurring we should see this would rightfully incur increased 500s. Returning a "denied" decision here would be a lie as the subscriber will never be allowed to successfully make this request.